far/copy.cpp

```diff
@@ -362,11 +362,15 @@ static string GenerateName(string_view const Name, string_view const Path)
 	const auto BaseSize = Result.size() - Name.size();
 	const auto NameExt = name_ext(Name);
 
+	auto GenNameFormat = Global->Opt->CMOpt.GenNameFormat.toString();
+	if (!contains(GenNameFormat, L"{0}"s))
+		GenNameFormat = Global->GenNameFormat;
+
 	// file (2).ext, file (3).ext and so on
 	for (int i = 2; os::fs::exists(Result); ++i)
 	{
 		Result.resize(BaseSize);
-		append(Result, NameExt.first, L" ("sv, str(i), L')', NameExt.second);
+		append(Result, format(GenNameFormat, i, NameExt.first, NameExt.second));
```

Comment:
fomat will throw is the format string is invalid.
And since it's defined by the user it will become invalid one day.

Please add try/catch, the same as here:
https://github.com/FarGroup/FarManager/blob/c4fe9ab36097de36a052e5316321c45e2a9d6bf1/far/execute.cpp#L459-L465